app/resources/v1/votes.js

```diff
@@ -6,25 +6,106 @@ const logger = require('../../../lib/logger.js')()
 exports.get = async function (req, res) {
   let ballot
   try {
-    ballot = await Vote.findAll({
-      where: {
-        voterId: {
-          [Sequelize.Op.is]: null
-        }
-      },
-      order: Sequelize.literal('random()'),
-      limit: 1
-    })
+    if (!req.user) {
+      ballot = await Vote.findAll({
+        where: {
+          voterId: {
+            [Sequelize.Op.is]: null
+          }
+        },
+        order: Sequelize.literal('random()'),
+        limit: 1
+      })
+    } else {
+      ballot = await Vote.findAll({
+        // where (submitted does not contain the user's auth0 ID, or is empty) AND (voterId is null)
+        where: {
+          [Sequelize.Op.and]: [
+            {
+              [Sequelize.Op.or]: [
+                {
+                  [Sequelize.Op.not]: {
+                    submitted: {
+                      [Sequelize.Op.contains]: [req.user.sub]
+                    }
+                  }
+                },
+                {
+                  submitted: {
+                    [Sequelize.Op.is]: null
+                  }
+                }
+              ]
+            },
+            {
+              voterId: {
+                [Sequelize.Op.is]: null
+              }
+            }
+          ]
+        },
+        order: Sequelize.literal('random()'),
+        limit: 1
+      })
+    }
   } catch (error) {
     logger.error(error)
     res.status(500).json({ status: 500, msg: 'Error fetching ballot.' })
     return
   }
+
+  if (ballot && ballot.length === 0) {
+    // no eligible streets remaining
+    res
+      .status(204)
+      .json({ status: 204, msg: 'all eligible streets have been voted on' })
+  }
+
   const payload = { ballot }
 
   res.status(200).json(payload)
 }
 
+exports.put = async function (req, res) {
+  const authUser = req.user || {}
+  const { id, comment } = req.body
+
+  if (!authUser.sub) {
+    res.status(401).json({ status: 401, msg: 'Please provide user ID.' })
+    return
+  }
+
+  let user
+
+  try {
+    user = await User.findOne({ where: { auth0Id: authUser.sub } })
+  } catch (error) {
+    logger.error(error)
+    res.status(500).json({ status: 500, msg: 'Error finding user.' })
+    return
+  }
+
+  if (!user) {
+    res.status(403).json({ status: 403, msg: 'User not found.' })
+    return
+  }
+
+  const ballot = await Vote.findOne({
+    where: { id: id, voter_id: user.auth0Id }
+  })
+
+  if (!ballot) {
+    res.status(403).json({ status: 403, msg: 'Ballot not found.' })
+  }
+
+  if (comment) {
+    ballot.comment = comment
```

Comment:
This is going to be free-form user-submitted content, so we need some guardrails in place. Do we have:
- input sanitation (preventing SQL injection, etc) (does Sequelize do this out of-the-box or do we need to provide our own?)
- input constraints (I think we need to specify a maximum length: maybe tweet length, 280 characters? It'd be best to add this constraint to the the table.)

Reply:
The `STRING` data type is limited to 255 characters by default, I'll update it to 280. 

From what I'm [reading](https://stackoverflow.com/questions/44272631/how-to-escape-in-sequelize), Sequelize does escape values when composed normally, however if we manually compose SQL we should use [sequelize.escape](https://sequelize.org/v5/class/lib/sequelize.js~Sequelize.html#instance-method-escape).